pulser-core/pulser/noise_model.py

```diff
@@ -422,6 +423,17 @@ def _validate_parameters(param_vals: dict[str, Any]) -> None:
                 comp = "a boolean"
             if not is_valid:
                 raise ValueError(f"'{param}' must be {comp}, not {value}.")
+            if param == "samples_per_run" and value != 1:
+                with warnings.catch_warnings():
+                    warnings.filterwarnings("always")
```

Comment:
Maybe "once" is enough
```suggestion
                    warnings.filterwarnings("once")
```

Reply:
Thinking about it some more, I'm not sure it even needs to be this loud. A standard `DeprecationWarning` with the filter might be enough and it would be easier to handle in the tests (ie you could use module / function level `pytest.mark.filerwarnings` instead of a bunch of `with deprecated_call()`). Wdyt?